components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/provider/ApplicationPermissionProvider.java

```diff
@@ -0,0 +1,26 @@
+package org.wso2.carbon.identity.application.mgt.provider;
+
+import org.wso2.carbon.identity.application.common.IdentityApplicationManagementException;
+import org.wso2.carbon.identity.application.common.model.ApplicationPermission;
+import org.wso2.carbon.identity.application.common.model.PermissionsAndRoleConfig;
+
+import java.util.List;
+
+/**
+ * Interface to provide application permissions.
+ */
+public interface ApplicationPermissionProvider {
+
+    void renameAppPermissionPathNode(String oldName, String newName) throws IdentityApplicationManagementException;
```

Comment:
```suggestion
    void renameAppPermissionName(String oldName, String newName) throws IdentityApplicationManagementException;
```

let's not use any registry references in the interface methods.

Reply:
Fixed: https://github.com/wso2/carbon-identity-framework/commit/a55181d5fdd11acf6f23808651ab50253ad85368